libbeat/outputs/elasticsearch/elasticsearch.go

```diff
@@ -61,6 +61,21 @@ type callbacksRegistry struct {
 // XXX: it would be fantastic to do this without a package global
 var connectCallbackRegistry = newCallbacksRegistry()
 
+type licenseCallbackHolder struct {
+	callback connectCallback
+	mutex    sync.RWMutex
+}
+
+var licenseCheck = licenseCallbackHolder{}
+
+// RegisterLicenseCallback register a global license callback, this enforce that all Elasticsearch
+// clients do the check.
+func RegisterLicenseCallback(cb connectCallback) {
```

Comment:
I really think we should avoid putting a hook like this into the OSS code. Could we change the registry to support ordering/priority in some way? Then the license callback can register with the highest priority and get invoked first and all the other callbacks get invoked afterwards

Reply:
Yeah that makes more sense in that case, let me make the change.